src/haskell/Compress_turbomack.hs

```diff
@@ -0,0 +1,18 @@
+import Data.Char (chr)
+import System.IO (getLine)
+
+count :: String -> [(Int, Char)]
+count str = foldr insertChar [] str
+  where
+    insertChar c [] = [(1, c)]
+    insertChar c ((count, char) : tail) =
+        if c == char
+            then (count + 1, char) : tail
+            else (1, c) : (count, char) : tail
+
+compress :: String -> String
+compress orig =
+    (\(c, v) -> [chr (c + 48), v]) =<< count orig
```

Comment:
this trick should work fine for sequences of 1-9 repeated characters, after that it won't work since it's just does shift in ascii character. This is sort of perf optimization. There are no tests that would clarify if this is acceptable or not so lazy programmer won.

Reply:
For now, I'm more than happy and impressed with the different solutions and ways people are approaching this problem.

My secret wish is to inspire non-Haskell (or other languages) engineers how others would do it and perhaps show different ideas and perhaps even translate them across.

Thank you Marek for your time and effort! I appreciate it!